OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -27,99 +27,149 @@ public class ScreenMapInfo : ITraitInfo
 
 	public class ScreenMap : IWorldLoaded
 	{
-		ScreenMapInfo info;
+		class Partitioned<T>
```

Comment:
Where would be a good place for it to live? I might rename it to ```SpatiallyPartitioned``` while I'm at it if it's going to be more visible.